Configuration.mk

```diff
@@ -28,8 +28,33 @@ KERNEL_HEAP_SIZE ?= 1024
 # PACKAGE_NAME is used to identify the application for IPC and for error reporting
 PACKAGE_NAME ?= $(shell basename "$(shell pwd)")
 
-# Tock supported architectures
-TOCK_ARCHS ?= cortex-m0|arm-none-eabi cortex-m3|arm-none-eabi cortex-m4|arm-none-eabi
+# Tock supported architectures.
+#
+# Each app will be compiled for each of the architectures.
+TOCK_ARCHS ?= cortex-m0 cortex-m3 cortex-m4 rv32imac
+
+# Tock app targets.
+#
+# This is a list of all of the different targets to build an app for which will
+# all be bundled into a TAB. This allows us to build an app for any board Tock
+# supports, and wait until a TAB is installed onto a board to figure out which
+# specific binary that hardware platform needs.
+#
+# Each entry is itself a list:
+#
+# 1. The name of the architecture. This is used for naming generated files and
+#    variables in the Makefiles. It is generally just a human-readable name.
+# 2. (Optional) The name to use when creating the output file.
+# 3. (Optional) The address to use as the fixed start of flash.
+# 4. (Optional) The address to use as the fixed start of RAM.
+#
+#  rv32imac|rv32imac.0x20040040.0x80002400|0x20040040|0x80002400 # RISC-V for HiFive1b
+#  rv32imac|rv32imac.0x40430060.0x80004000|0x40430060|0x80004000 # RISC-V for arty-e21
+TOCK_TARGETS ?= cortex-m0\
+                cortex-m3\
+                cortex-m4\
+                rv32imac|rv32imac.0x20040040.0x80002400|0x20040040|0x80002400\
+                rv32imac|rv32imac.0x40430060.0x80004000|0x40430060|0x80004000
```

Comment:
Can OT be added as well?

Reply:
Yes of course.

Reply:
What addresses should I add?

Reply:
FLASH (rx) : ORIGIN = 0x20030040, LENGTH = 32M
  SRAM (rwx) : ORIGIN = 0x10002D00, LENGTH = 512K